src/users.js

```diff
@@ -33,96 +39,117 @@ var route = function (request, response) {
-                    bcrypt.compare(credentials['password'], data['password'], function (err, res) {
-                        if (res == true) {
-                            var generated_token = generate_token();
-                            //We get the array tokens in the server cache
-                            myCache.get("tokens", function (err, value) {
-                                if (!err) {
-                                    if (value == undefined) {
-                                        //The array in the cache does not exist : Init of the array containing the tokens
-                                        myCache.set("tokens", [generated_token], function (err) {
-                                            if (err) {
-                                                response.writeHead(201, { "Content-Type": "application/json" });
-                                                response.end();
-                                            }
-                                        });
-                                    } else {
-                                        //The array tokens already exists in the cache
-                                        var tokens = value;
-                                        tokens.push(generated_token);
-                                        //Already a token in the server, we add the new generated token in the tokens cache array
-                                        myCache.set("tokens", tokens, function (err) {
-                                            if (err) {
-                                                response.writeHead(201, { "Content-Type": "application/json" });
-                                                response.end();
-                                            }
-                                        });
-                                    }
-                                }
-                            });
-                            response.writeHead(200, { "Content-Type": "application/json" });
-                            response.end(JSON.stringify({ token: generated_token }));
-                        } else {
-                            response.writeHead(201, { "Content-Type": "application/json" });
-                            response.end();
-                        }
-                    });
-                } else {// The username does not exist in the DB
-                    //FIXME : If no user in database, which status code should I return ?
-                    response.writeHead(201, { "Content-Type": "application/json" });
-                    response.end();
-                }
+            //See usage of bcrypt library : https://www.npmjs.com/package/bcrypt
+            bcrypt.hash(credentials['password'], saltRounds, function (err, hash) {
+                credentials['password'] = hash;
+                api.addUser(credentials, function (data) {
+                    if (data == false) {//User already existing
```

Comment:
Please use triple comparator (===) instead of double ones (==) for boolean and undefined.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators